CMakeLists.txt

```diff
@@ -27,6 +27,76 @@ ELSE()
     SET(VER_BUILD 0)
 ENDIF()
 
+# Configuration of compilators
+## Compiler flags to add
+### Flags for GCC and Clang
+set(GNU_FLAGS_WARNING_YES "-Wall -Wextra -Wformat=2 -Wpedantic -Wshadow -Wmissing-declarations -Wfloat-equal")
```

Comment:
Please, whatever you do and as a good practice, provide comments of what you are doing. E.g.:
# Adding standard warning compiler warnings 
etc ...

Secondly, what are you forcing every compile type flags? Could you tell me why the default flags used by CMake don't suit you?
http://stackoverflow.com/questions/24767450/using-cmake-how-do-you-determine-the-default-compiler-flags-per-build-type

Third, as also told in the link above, I'd rather use/set directly CMAKE_CXX_FLAGS_** based on the compile type rather than setting yet other variables there. But the actual use of all this is yet to be proven.